test_runner/regress/test_migrations.py

```diff
@@ -15,7 +15,7 @@ def test_migrations(neon_simple_env: NeonEnv):
     endpoint.respec(skip_pg_catalog_updates=False)
     endpoint.start()
 
-    num_migrations = 10
+    num_migrations = 11
```

Comment:
Testing is also one of my concerns. Basically we now only test that migrations can apply on the empty cluster. It's something, but it's not that interesting

What would be more interesting is to prepare some 'messy' cluster, i.e. create a few roles, databases (template database, without allowed connections, ideally an invalid one too), do some grants or maybe install extensions, etc. And only then run all migrations

I think that currently, it's not possible because we always apply till the latest migration. We can probably add a new optional spec attribute, e.g. `migration_target`. If it's not set, then apply till the latest. If it's set, then apply only till the specified migration. After that, it would be possible to do this test via using `respec`

Reply:
I wonder if we could solve this with fail points instead of using a spec. I will work on the testing side of this.

Reply:
I guess compute_ctl doesn't support failpoints right now. Perhaps we should support it.